aspnet/migration/aspnetcoremigration.rst

```diff
@@ -0,0 +1,190 @@
+    "version": "1.0.0-*",
+    "imports": "portable-net45+wp80+win8+wpa81+dnxcore50"
+  }
+},
+"scripts": {
+  "postpublish": "dotnet publish-iis --publish-folder %publish:OutputPath% --framework %publish:FullTargetFramework%"
+}
+
+Additionally, you must turn on server garbage collection in ``project.json`` or ``app.config`` when on full .NET framework.
+
+Changes in MVC
+--------------
+
+- Set option in ``project.json`` to preserve the compilation context if you are doing views.
+- You no longer need to reference the Tag Helper package, it's now referenced by ``Razor`` by default.
+
+There are changes that simplify controller discovery:
+There is a new ``Controller`` attribute that can be used to mark a class and their descendants as controllers.
+Classes whose name doesn't end in ``Controller`` and derive from a base class that ends in ``Controller`` are no longer considered controllers. In this scenario the ``[Controller]`` attribute must be applied to the ``Controller`` class itself or to the base class.
+
+We now consider a type to be a controller if all of the following rules apply:
+- The type is a public, concrete, non open generic class.
+- [NonController] is not applied to any type of the hierarchy.
+- The type name ends with ``Controller``, or if the ``[Controller]`` attribute is applied to the type or to one of its ancestors.
+- It's important to note that if [NonController] is applied anywhere in the type hierarchy the discovery conventions will never consider that type or its descendants to be a controller. ``[NonController]`` takes precedence over ``[Controller]``.
+
+
+ViewComponents changes
+----------------------
+
+-The Sync APIs have been removed.
+-``InvokeAsync`` takes an anonynmous object instead of params.
+-``web`` is no longer a thing, use dotnetrun or ``dotnet <dll name>`` at the command prompt instead.
+
+Mapping commands to tools
+-------------------------
+
+The following commands are now tools: Secret manager, watch, sqlcache. You can configure these in the ``tools`` section in the ``project.json`` file found in the root of your project. 
+
+Configuration
+-------------
+
+``IConfigurationSource`` has been introduced to represent the settings/configuration which is used to Build() an IConfigurationProvider. It is no longer possible to access the provider instances from IConfigurationBuilder only the sources. This is intentional, but may cause loss of functionality as you can longer do things like explicitly call Load() on the provider instances.
+
+FileConfigurationProvider base class has been introduced as a common root for Json/Xml/Ini providers. This allows the ability to specify an IFileProvider on the source which will be used to read the file instead of explicitly using File.Open. The side effect of this change is that absolute paths are no longer supported, the file path must be relative to the base path of the IConfigurationBuilder's basepath or the IFileProvider if specified.
```

Comment:
> which will be used

Nit: Maybe drop that to make the sentence more concise.

This allows the ability to specify an IFileProvider on the source to read the file instead of explicitly using File.Open.

> supported, the

Nit: Two complete thoughts (sentences). Recommend a period and new sentence there.